src/vs/workbench/contrib/terminal/browser/xterm/quickFixAddon.ts

```diff
@@ -101,6 +96,7 @@ export class TerminalQuickFixAddon extends Disposable implements ITerminalAddon,
 			});
 		}
 		this._quickFixService.onDidRegisterProvider(result => this.registerCommandFinishedListener(convertToQuickFixOptions(result)));
+		this._quickFixService.onDidRegisterCommandSelector(selector => this.registerCommandSelector(selector));
```

Comment:
You might have a race condition here, I think you also need to iterate over any command selectors that exist at the time `TerminalQuickFixAddon` is created?